packages/components/src/spinner/index.js

```diff
@@ -1,8 +1,54 @@
+/**
+ * External dependencies
+ */
+import styled from '@emotion/styled';
+
 /**
  * Internal dependencies
  */
-import { StyledSpinner } from './styles/spinner-styles';
+import { COLORS, CONFIG } from '../utils';
+
+/**
+ * WordPress dependencies
+ */
+import { SVG, Circle } from '@wordpress/primitives';
+
+// A dedicated wrapper allows us to apply width and shadow color based on config values
+// The SVG can only have a centered stroke so a stacked box-shadow accomplishes a matching 1.5px border
+export const StyledSpinner = styled.span`
+	width: ${ CONFIG.spinnerSize }px;
+	height: ${ CONFIG.spinnerSize }px;
+	box-shadow: inset 0 0 0 0.75px ${ COLORS.gray[ 300 ] },
+		0 0 0 0.75px ${ COLORS.gray[ 300 ] };
+`;
 
 export default function Spinner() {
-	return <StyledSpinner className="components-spinner" />;
+	const viewBox = `0 0 ${ CONFIG.spinnerSize } ${ CONFIG.spinnerSize }`;
+	const radius = Number( CONFIG.spinnerSize ) / 2;
+	const strokeDasharray = [
+		Number( CONFIG.spinnerSize ),
+		Number( CONFIG.spinnerSize ) * 10,
+	];
+
+	return (
+		<StyledSpinner className="components-spinner">
+			<SVG
+				viewBox={ viewBox }
+				xmlns="http://www.w3.org/2000/svg"
+				className="components-spinner__track"
+				role="img"
+				aria-hidden="true"
```

Comment:
Using `role="img"` and `aria-hidden="true"` seems contradictory: on one hand we're telling the browser that this is a meaningful image (for which we don't provide alt text), on the other we're telling the browser (and any assistive technology) to ignore this element.

I'm not an expert in a11y patterns, but from a quick look, it seems like a good combination could be using `role="alert"` and `aria-live="assertive"` ([example from carbon design system](https://www.carbondesignsystem.com/components/loading/accessibility)). If we don't want to investigate this route too much, in my opinion we should at least:

- remove `aria-hidden="true"`
- keep the `role="img"` attribute
- provide an accessible label. According to [this article](https://css-tricks.com/accessible-svgs/), the best way to achieve this would be to add a `<title id="uniqueTitleID">The title of the SVG</title>` tag inside the SVG element, and then to add `aria-labelledby="uniqueTitleID"` to the SVG element itself

Reply:
These are excellent points, but as this PR seeks only to update the design may I suggest we investigate a11y in a dedicated issue?

Reply:
That's a fair point.

Maybe a good interim solution would be to remove `aria-hidden="true"` and change the role to `role="presentation"`. We can then investigate the a11y in a follow-up PR as you suggested